crates/sui-core/src/authority/authority_store.rs

```diff
@@ -738,6 +738,34 @@ impl AuthorityStore {
         Ok(())
     }
 
+    /// Insert objects directly into the object table, but do not touch other tables.
+    /// This is used in fullnode to insert objects from validators certificate handling response
+    /// in fast path execution.
+    /// This is best-efforts. If the object needs to be stored as an indirect object then we
+    /// do not insert this object at all.
+    pub(crate) fn fullnode_fast_path_insert_objects_to_object_store_maybe(
+        &self,
+        objects: &Vec<Object>,
+    ) -> SuiResult {
+        let mut write_batch = self.perpetual_tables.objects.batch();
+
+        for obj in objects {
+            let StoreObjectPair(store_object, indirect_object) =
+                get_store_object_pair(obj.clone(), self.indirect_objects_threshold);
+            // Do not insert to store if the object needs to stored as indirect object too.
+            if indirect_object.is_some() {
+                continue;
```

Comment:
this shouldn't ever happen, so lets log an warn! if it does

Reply:
@phoenix-o what if the object already existed, as an indirect object. this would not decrement the ref count, right? this shouldn't happen, but it could in theory.

Reply:
in person discussion was that this would only happen in very implausible situations, and the only consequence would be an orphaned clock object, so we will ignore it